src/hssm/plotting/posterior_predictive.py

```diff
@@ -189,21 +191,98 @@ def _plot_posterior_predictive_2D(
 
     if plot_data:
         g.add_legend(
-            dict(zip(["predicted", "observed"], g.fig.axes[0].get_lines())),
+            dict(zip(["predicted", "observed"], g.figure.axes[0].get_lines())),
             title="",
             label_order=["observed", "predicted"],
         )
 
     if title:
-        g.fig.subplots_adjust(top=0.9)
-        g.fig.suptitle(title)
+        g.figure.subplots_adjust(top=0.9)
+        g.figure.suptitle(title)
 
     g.set_xlabels(xlabel)
     g.set_ylabels(ylabel)
 
     return g
 
 
+def _process_linestyles_pp(linestyles):
+    if isinstance(linestyles, str):
+        return [linestyles, linestyles]
+
+    if isinstance(linestyles, list) or isinstance(linestyles, tuple):
+        if all(isinstance(ls, str) for ls in linestyles):
+            if len(linestyles) == 1:
+                return [linestyles[0], linestyles[0]]
+            elif len(linestyles) == 2:
+                return linestyles
+            else:
+                raise ValueError(
+                    "The `linestyles` argument must be a string or a list of length 1 "
+                    "or 2."
+                )
+        else:
+            raise ValueError(
+                "The `linestyles` argument must be a string or a list of strings."
+            )
```

Comment:
Where do you stand on guard clauses?
```suggestion
    if isinstance(linestyles, (list, tuple)):
        linestyles = list(linestyles)
        if not all(isinstance(ls, str) for ls in linestyles):
            raise ValueError("The `linestyles` list must contain only strings.")
        if len(linestyles) in {1, 2}:
            return linestyles * 2 if len(linestyles) == 1 else linestyles
        raise ValueError("The `linestyles` list must have length 1 or 2.")
```